R/build-articles.R

```diff
@@ -102,36 +105,63 @@ render_rmd <- function(pkg,
                        toc = TRUE,
                        depth = 1L,
                        encoding = "UTF-8",
+                       render_asis = FALSE,
                        quiet = TRUE) {
 
   message("Building article '", output_file, "'")
   scoped_package_context(pkg$package, pkg$topic_index, pkg$article_index)
   scoped_file_context(depth = depth)
 
-  format <- build_rmarkdown_format(pkg, depth = depth, data = data, toc = toc)
-  on.exit(unlink(format$path), add = TRUE)
+  if (!render_asis)
+  {
+    format <- build_rmarkdown_format(pkg, depth = depth, data = data, toc = toc)
+    on.exit(unlink(format$path), add = TRUE)
+  }
 
-  path <- callr::r_safe(
-    function(...) rmarkdown::render(...),
-    args = list(
+  args <- if (render_asis) {
+    list(
+      input,
+      output_file = basename(output_file),
+      quiet = quiet,
+      envir = globalenv()
+    )
+  } else
+  {
+
+    list(
       input,
       output_format = format$format,
       output_file = basename(output_file),
       quiet = quiet,
       encoding = encoding,
       envir = globalenv()
-    ),
+    )
+  }
+
+  path <- callr::r_safe(
+    function(...) rmarkdown::render(...),
+    args = args,
     show = !quiet
   )
-  update_rmarkdown_html(path, strip_header = strip_header, depth = depth)
+
+  if (!render_asis)
+  {
+    update_rmarkdown_html(path, strip_header = strip_header, depth = depth)
+  }
```

Comment:
Since this function is called primarily for its side effects can you please return `invisible(pkg)` here?